Framework/CurveFitting/test/FunctionFactoryConstraintTest.h

```diff
@@ -204,6 +212,11 @@ class FunctionFactoryConstraintTest : public CxxTest::TestSuite {
 
     funa->setParameter("a1", 11);
     TS_ASSERT_EQUALS(c1->check(), 0);
+
+    funa->setConstraintPenaltyFactor("a1", 18.4);
+    TS_ASSERT_EQUALS(funa->asString(),
+                     "name=FunctionFactoryConstraintTest_FunctA,a0=-1,"
+                     "a1=11,constraints=(0<a0<0.2,10<a1,penalty=18.4)");
```

Comment:
A test is needed to check that FunctionFactory can set penalty factors from string inputs, for example:
```
funb = FunctionFactory::Instance().createInitialized(
        "name=FunctA,a0=-1,a1=1.1,constraints=(0<a0<0.2,penalty=11.2,a1>10, penalty=10.2)");
```
Various possible cases should be considered.

Reply:
Added a few unit tests that check this from both the penalty being in the string input to the function (as in your example) and when you add constraints (e.g. `funa->addConstraints("0<a0<0.2,penalty=8,a1>10");`).  The latter required changes to IFunction::addConstraints.  I think this covers all the ways you can add constraints.